numba_dpex/experimental/launcher.py

```diff
@@ -367,30 +477,101 @@ def codegen(cgctx, builder, sig, llargs):
             local_range_extents=index_space_values.local_range_extents,
         )
 
-        fn_body_gen.submit_and_wait(submit_call_args)
-
-        fn_body_gen.cleanup(kernel_bundle_ref=kbref, kernel_ref=kref)
+        eref = fn_body_gen.submit(submit_call_args)
+        device_event = wrap_event_reference(cgctx, builder, eref)
```

Comment:
Why is this required for `call_kernel` (synchronous) case?

Reply:
I wanted to reuse the intrinsic for both `call_kernel` and `call_kernel_async`